x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -489,7 +489,7 @@ public void testStringAsIndexPattern() {
             clusterAndIndexAsIndexPattern(command, "cluster:index");
             clusterAndIndexAsIndexPattern(command, "cluster:.index");
             clusterAndIndexAsIndexPattern(command, "cluster*:index*");
-            clusterAndIndexAsIndexPattern(command, "cluster*:<logstash-{now/D}>*");
+            clusterAndIndexAsIndexPattern(command, "cluster*:<logstash-{now/D}*>");
```

Comment:
I think this is not valid. `*` must be within date math `<>`:

```
FROM "<index-{now/d{yyyy}}-*>" -> resolves to `index-2025-a`
FROM "<index-{now/d{yyyy}}->*" -> "reason": "Found 1 problem\nline 1:1: Unknown index [<index-{now/d{yyyy}}->*]"
```
when running ES locally

Reply:
We also test following patterns:

https://github.com/elastic/elasticsearch/blob/ea3163e67daa29e9e85a67bf22dceaa78303f386/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java#L487-L488

I do not think they are valid. Please confirm

Reply:
I think we should not alter the test case, but add another one with the `*` inside the date-math pattern + add a comment that the wildcard outside is not permitted.

If anything, the parser should reject `<logstas-{now/D}*>` (with the wildcard inside). But the parser itself can be more lenient, because for invalid patterns we'll just fail to find matching indices in the field caps call.

So it's safe to be a bit more permissive (and document that via tests) than to reject too many things accidentally.